dot/state/block.go

```diff
@@ -258,7 +280,13 @@ func (bs *BlockState) GetBlockHashesBySlot(slotNum uint64) ([]common.Hash, error
 	}
 
 	descendants, err := bs.bt.GetAllDescendants(highestFinalisedHash)
-	if err != nil {
+	if errors.Is(err, blocktree.ErrNodeNotFound) {
+		var err2 error
+		descendants, err2 = bs.GetAllDescendants(highestFinalisedHash)
```

Comment:
Actually, you're not retrieving the highest finalized hash from the disk because:

- The method `bs.GetAllDescendants` is calling `blocktree.GetAllBlocksAtNumber` which stills uses the method `blocktree.getNode`, if it does not find the given hash in the in-memory block tree it returns an empty hash slice instead of an error

`blocktree.GetAllBlocksAtNumber`: https://github.com/ChainSafe/gossamer/blob/043f5eb77f0468f61e5721b8833b032df9f84562/lib/blocktree/blocktree.go#L112

Reply:
I don't need to retrieve highest finalised hash. I care for its descendants.

Reply:
> I care for its descendants.

I get the point! 

However, given the current implementation of the `GetAllBlocksAtNumber` method (used by `blockState. GetAllDescendants`), the problem is still not solved, the reason:

- `GetAllBlocksAtNumber` receives the given hash and keeps using the `blocktree.getNode` to retrieve it from the blocktree, if the given hash is not found in the block tree instead of an error it returns an empty slice which is worse because it does not causes a problem but passes without any warns

So to reiterate, in this PR the problem still happening but this time it will simply pass without errors due to the `GetAllBlocksAtNumber` you are using, I would like to propose

Reply:
re-did this with new approach, that seems to fix it.